Documentation/gettingstarted/bgp.rst

```diff
@@ -0,0 +1,168 @@
+.. only:: not (epub or latex or html)
+
+    WARNING: You are looking at unreleased Cilium documentation.
+    Please use the official rendered version released here:
+    https://docs.cilium.io
+
+.. _bgp:
+
+**********
+BGP (beta)
+**********
+
+BGP provides a way to advertise routes using traditional networking protocols
+to allow Cilium-managed services to be accessible outside the cluster.
+
+This document explains how to configure Cilium's native support for announcing
+``LoadBalancer`` IPs of ``Services`` via BGP. It leverages `MetalLB's
+<https://metallb.universe.tf/>`_ simple and effective implementation of IP
+allocation and the minimal BGP protocol support to do this. The configuration
+for Cilium is the same as MetalLB's configuration.
+
+Specifically, if a ``Service`` of type ``LoadBalancer`` is created, Cilium will
+allocate an IP for it from a specified pool. Once the IP is allocated, the
+Agents will announce via BGP depending on the ``Service``'s
+``ExternalTrafficPolicy``. See MetalLB's `documentation
+<https://metallb.universe.tf/usage/#bgp>`_ on this specific topic.
+
+.. include:: ../beta.rst
+
+Deploy Cilium
+=============
+
+.. include:: k8s-install-download-release.rst
+
+BGP support is enabled by providing the BGP configuration via a ConfigMap and
+by setting a few Helm values. Otherwise, BGP is disabled by default.
+
+Here's an example ConfigMap:
+
+.. code-block:: yaml
+
+   apiVersion: v1
+   kind: ConfigMap
+   metadata:
+     name: bgp-config
+     namespace: kube-system
+   data:
+     config.yaml: |
+       peers:
+         - peer-address: 10.0.0.1
+           peer-asn: 64512
+           my-asn: 64512
+       address-pools:
+         - name: default
+           protocol: bgp
+           addresses:
+             - 192.0.2.0/24
+
+Here are the required Helm values:
+
+.. parsed-literal::
+
+   helm install cilium |CHART_RELEASE| --set bgp.enabled=true --set bgp.announce.lbIP=true
+
+Verify that Cilium Agent pod is running.
+
+.. code-block:: shell-session
+
+   $ kubectl -n kube-system get pods -l k8s-app=cilium
+   NAME           READY   STATUS    RESTARTS   AGE
+   cilium-5ngzd   1/1     Running   0          3m19s
+
+Create LoadBalancer and backend pods
+====================================
+
+Apply the following ``LoadBalancer`` ``Service`` and its corresponding
+backends:
+
+.. code-block:: yaml
+
+   apiVersion: v1
+   kind: Service
+   metadata:
+     name: test-lb
+   spec:
+     type: LoadBalancer
+     ports:
+     - port: 80
+       targetPort: 80
+       protocol: TCP
+       name: http
+     selector:
+       svc: test-lb
+   ---
+   apiVersion: apps/v1
+   kind: Deployment
+   metadata:
+     app: nginx
+   spec:
+     selector:
+       matchLabels:
+         svc: test-lb
+     template:
+       metadata:
+         labels:
+           svc: test-lb
+       spec:
+         containers:
+         - name: web
+           image: nginx
```

Comment:
It looks like nginx is only used as example app in other page, do we have a demo app that is good for this? If not, I think [podinfo](https://github.com/stefanprodan/podinfo) could be a good candidate. Just a thought, and I guess here it doesn't matter all that much as we are not doing blue/green deployments or anything like that, but may a simple app that has Cilium logo would be nice.